### PR TITLE
MacOS improvements

### PR DESCRIPTION
**Pull request by atineoSE**
This PR adds some improvements related to running the OpenAdapt framework on macOS (tested on macOS Ventura), namely:
- missing dependencies requirements
- add instructions to set up permissions for proper execution of framework

See inline comments for further details.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,6 @@ cache
 
 # db
 *.db
+
+# Packages in editable mode
+src
```
Comment (atineoSE) [inline .gitignore:18]: Do not keep track on git of dependency checked out in editable mode (they will go under `src` folder)

File: requirements.txt [hunk 2]
```diff
@@ -12,8 +12,10 @@ mss==6.1.0
 pandas==2.0.0
 pygetwindow==0.0.9
 pyinstaller
+setuptools-lint
+sphinx
```
Comment (atineoSE) [inline requirements.txt:16]: Missing dependencies, as tested on a fresh environment on macOS

File: requirements.txt [hunk 3]
```diff
@@ -12,8 +12,10 @@ mss==6.1.0
...
 pywin32==306; sys_platform == 'win32'
-git+https://github.com/abrichr/pynput.git
+-e git+https://github.com/abrichr/pynput.git#egg=pynput
```
Comment (atineoSE) [inline requirements.txt:18]: Checkout pynput fork in editable mode, for easy development

File: .gitignore (hunk 1 above)
Comment (abrichr) [reply .gitignore:18]: @atineoSE can you please clarify what is/are the command(s) you run to generate the `src` directory?

File: requirements.txt (hunk 3 above)
Comment (abrichr) [reply requirements.txt:18]: Thank you!

**Comment by abrichr** [comment]
Thank you @atineoSE !

What should we do about the instructions here? https://github.com/MLDSAI/OpenAdapt#troubleshooting

File: .gitignore (hunk 1 above)
Comment (atineoSE) [reply .gitignore:18]: @abrichr this was generated when importing the pynput dependency as editable framework. The pynput package is placed under `./src/pynput`

**Comment by atineoSE** [comment]
Good point about the troubleshooting section, @abrichr. I have suggested a change at commit [f6917b5](https://github.com/MLDSAI/OpenAdapt/pull/120/commits/f6917b5f8ff22b69cb94b7c47d90c09ff9c700e4), to redirect the reader to the permissions page. I haven't seen those terminal error messages, I got the system alerts instead.

**Comment by atineoSE** [comment]
@abrichr I have rebased over latest changes and fixed conflicts. Please, let me know if you need anything else before this can be merged. Thanks!

**Comment by abrichr** [comment]
Thank you @atineoSE ! Will look at it ASAP 🙏

File: README.md [hunk 4]
```diff
@@ -33,6 +33,10 @@ alembic upgrade head
 pytest
 ```
 
+## Permissions
+
+See how to set up system permissions on macOS [here](./permissions_in_macOS.md).
+
```
[the remark this answers is not in the document]
Comment (abrichr) [reply README.md:41]: @atineoSE what do you think about including the contents of this file in the existing `README.md` in a separate top-level section?

File: README.md [hunk 5]
```diff
@@ -178,25 +182,10 @@ If you're interested in getting paid for your work, please mention it in your Pu
...
-Solution:
-https://stackoverflow.com/a/69673312
-
-```
-Settings -> Security & Privacy
-Click on the Privacy tab
-Scroll and click on the Accessibility Row
-Click the +
-Navigate to /System/Applications/Utilities/ or wherever the Terminal.app is installed
-Click okay.
```
Comment (abrichr) [inline README.md:198]: @atineoSE can you please clarify the reason behind removing this section?

Comment (abrichr) [reply README.md:198]: I would like to merge the rest ASAP so I will revert this for now. Please feel free to open up a separate PR to continue the discussion! 🙏

**Pull request merged**